Geometry/CaloTopology/interface/EcalBarrelTopology.h

```diff
@@ -7,22 +7,23 @@
 #include "DataFormats/EcalDetId/interface/EBDetId.h"
 #include "Geometry/CaloTopology/interface/CaloSubdetectorTopology.h"
 #include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
 #include "FWCore/Framework/interface/ESHandle.h"
 
 class EcalBarrelTopology final : public CaloSubdetectorTopology
 {
 
- public:
+public:
   /// create a new Topology
-  EcalBarrelTopology(): theGeom_(nullptr) {};
+ EcalBarrelTopology(): theGeom_(nullptr) {};
 
   /// virtual destructor
   ~EcalBarrelTopology() override { }  
 
   /// create a new Topology from geometry
-  EcalBarrelTopology(edm::ESHandle<CaloGeometry> theGeom) : theGeom_(std::move(theGeom))
-    {
-    }
+  EcalBarrelTopology(edm::ESHandle<CaloGeometry> theGeom) {
+    theGeom_ = (theGeom.product())->getSubdetectorGeometry(DetId::Ecal,EcalBarrel);
```

Comment:
[just checking diffs wrt what I signed before]: since this was improved, I guess even shorter expression could be ```theGeom->getSubdetectorGeometry(DetId::Ecal,EcalBarrel);```